presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java

```diff
@@ -443,8 +442,12 @@ private static ConnectorPageSource createBatchOrcPageSource(
     {
         OrcDataSource orcDataSource = null;
         try {
-            FileSystem fileSystem = hdfsEnvironment.getFileSystem(user, path, configuration);
-            FSDataInputStream inputStream = hdfsEnvironment.doAs(user, () -> fileSystem.open(path));
+            ExtendedFileSystem fileSystem = hdfsEnvironment.getFileSystem(user, path, configuration);
```

Comment:
While you are at it, let's clean it up a bit further. If you look at the call site of this function, you'll find that it calls `fileSystem.getFileStatus` to get file size then passes `fileSize` as a parameter to this function. Now that we are calling `getFileStatus` here for modification time as well, you can just remove the `getFileStatus` call from line 240-247.

Reply:
Done